rdtools/plotting.py

```diff
@@ -256,14 +259,61 @@ def soiling_rate_histogram(soiling_info, bins=None):
 
     soiling_summary = soiling_info['soiling_interval_summary']
     fig, ax = plt.subplots()
-    ax.hist(100.0 * soiling_summary.loc[soiling_summary['valid'], 'soiling_rate'],
-            bins=bins)
+    ax.hist(100.0 * soiling_summary.loc[soiling_summary['valid'],
+                                        'soiling_rate'], bins=bins)
     ax.set_xlabel('Soiling rate (%/day)')
     ax.set_ylabel('Count')
 
     return fig
 
 
+def tune_clip_filter_plot(power_ac, clipping_mask, display_web_browser=True):
+    """
+    This function allows the user to visualize a clipping filter in
+    a matplotlib plot, after tweaking the function's different
+    hyperparameters. The plot can be zoomed in on, for an in-depth look at
+    clipping in the AC power time series.
+
+    Parameters
+    ----------
+    power_ac : pd.Series
+        AC power in Watts. Index of the Pandas series is a Pandas
+        datetime index.
+    clipping_mask : pd.Series
+        Pandas series of booleans, where good data periods (no clipping)
+        are marked as True, and omitted-data periods where clipping
+        occurs are marked as False.
+    display_web_browser : boolean, default True
+        When set to True, the Plotly graph is displayed in the
+        user's web browser.
+
+    Returns
+    ---------
+    Interactive Plotly graph, with the masked time series for clipping.
+    """
+    # Get the names of the series and the datetime index
+    column_name = power_ac.name
+    if column_name is None:
+        column_name = 'power_ac'
+        power_ac = power_ac.rename(column_name)
+    index_name = power_ac.index.name
+    if index_name is None:
+        index_name = 'datetime'
+        power_ac = power_ac.rename_axis(index_name)
+    # Visualize the power_ac time series, delineating clipping periods
+    # using the clipping_mask series. Use plotly to visualize.
+    df = pd.DataFrame(power_ac)
+    # Add the clipping mask as a column
+    df['clipping_mask'] = clipping_mask
+    df = df.reset_index()
+    fig = px.scatter(df, x=index_name, y=column_name, color='clipping_mask')
+    # If display_web_browser is set to True, the time series with clipping
+    # is rendered via the web browser.
+    if display_web_browser is True:
+        fig.show(renderer="browser")
+    return fig
```

Comment:
I'd suggest adjusting the legend. The current version can be confusing because clipping_filter=True actually means clipping not detected. Maybe the legend values can be changed True->not clipping and False->clipping?